scripts/release.js

```diff
@@ -31,35 +41,87 @@ const humanReadableTypes = {
 
-  // build may have generated a new src-docs/src/i18ntokens.json file, dirtying the git workspace
-  // it's important to track those changes with this release, so determine the changes and write them
-  // to src-docs/src/i18ntokens_changelog.json, comitting both to the workspace before running `npm version`
-  execSync(`npm run update-token-changelog -- ${versionTarget}`, execOptions);
+    // build may have generated a new src-docs/src/i18ntokens.json file, dirtying the git workspace
+    // it's important to track those changes with this release, so determine the changes and write them
+    // to src-docs/src/i18ntokens_changelog.json, comitting both to the workspace before running `npm version`
+    execSync(`npm run update-token-changelog -- ${versionTarget}`, execOptions);
 
-  // update package.json & package-lock.json version, git commit, git tag
-  execSync(`npm version ${versionTarget}`, execOptions);
+    // update package.json & package-lock.json version, git commit, git tag
+    execSync(`npm version ${versionTarget}`, execOptions);
+  }
 
-  // push the version commit & tag to upstream
-  execSync('git push upstream --tags', execOptions);
+  if (args.steps.indexOf('tag') > -1) {
+    // push the version commit & tag to upstream
+    execSync('git push upstream --tags', execOptions);
+  }
 
-  // prompt user for npm 2FA
-  const otp = await getOneTimePassword();
+  if (args.steps.indexOf('publish') > -1) {
+    // prompt user for npm 2FA
+    const otp = await getOneTimePassword();
 
-  // publish new version to npm
-  execSync(`npm publish --otp=${otp}`, execOptions);
+    // publish new version to npm
+    execSync(`npm publish --otp=${otp}`, execOptions);
+  }
 
-  // update docs, git commit, git push
-  execSync('npm run sync-docs', execOptions);
+  if (args.steps.indexOf('docs') > -1) {
+    // update docs, git commit, git push
+    execSync('npm run sync-docs', execOptions);
+  }
 }()).catch(e => console.error(e));
 
+function parseArguments() {
+  const parser = new argparse.ArgumentParser({
+    addHelp: true,
+    description: 'Tag and publish a new version of EUI',
+  });
+
+  parser.addArgument('--type', {
+    help: 'Version type; can be "major", "minor" or "patch"',
+    choices: Object.values(humanReadableTypes),
+  });
+
+  parser.addArgument('--dry-run', {
+    action: 'storeTrue',
+    defaultValue: false,
+    help: 'Dry run mode; no changes are made',
+  });
+
+  const allSteps = ['test', 'build', 'version', 'tag', 'publish', 'docs'];
+  parser.addArgument('--steps', {
+    help: 'Which release steps to run; a comma-separated list of values that can include "test", "build", "version", "tag", "publish" and "docs". If no value is given, all steps are run. Example: --steps=test,build,version,tag',
+    defaultValue: allSteps.join(','),
+  });
+
+  const args = parser.parseArgs();
+
+  // validate --steps argument
+  const steps = args.steps.trim().split(',');
```

Comment:
```suggestion
  const steps = args.steps.split(',').map(step => step.trim());
```

to handle whitespace anywhere in the list, not just at the front & end

Reply:
Because the string is coming from bash, it's worth noting that in order to pass a comma-separated list with spaces, it'd also have to be quoted, e.g. `--steps "test, build, version"`.

That said, I always appreciate a belt-and-suspenders safeguard. 😄